src/sparseml/transformers/utils/helpers.py

```diff
@@ -242,6 +227,255 @@ def resolve_sequence_length(config: AutoConfig) -> int:
+        )
+        recipe = os.path.join(model_path, RECIPE_NAME)
+        if os.path.isfile(recipe):
+            _LOGGER.info(f"Found recipe in the model_path: {recipe}")
+            return recipe
+        _LOGGER.debug(f"No recipe found in the model_path: {model_path}")
+        return None
+
+    elif model_path.startswith("zoo:"):
+        # model_path is a sparsezoo stub
+        return recipe_from_sparsezoo_stub(stub=model_path)
+
+    recipe = recipe_from_huggingface_model_id(model_path)[0]
+
+    if recipe is None:
+        _LOGGER.info("Failed to infer the recipe from the model_path")
+    return recipe
+
+
+def recipe_from_huggingface_model_id(
+    model_path: str, recipe_name: str = RECIPE_NAME
+) -> Tuple[Optional[str], bool]:
+    """
+    Attempts to download the recipe from the huggingface model id.
+
+    :param model_path: Assumed to be the huggingface model id.
+        If it is not, this function will return None.
+    :param recipe_name: The name of the recipe file to download.
+        Defaults to RECIPE_NAME.
+    :return: tuple:
+        - the path to the recipe file if found, None otherwise
+        - True if model_path is a valid huggingface model id, False otherwise
+    """
+    model_id = os.path.join(HUGGINGFACE_CO_URL_HOME, model_path)
+    request = requests.get(model_id)
+    if not request.status_code == 200:
+        _LOGGER.debug(
+            "model_path is not a valid huggingface model id. "
+            "Skipping recipe resolution."
+        )
+        return None, False
+
+    _LOGGER.info(
+        "model_path is a huggingface model id. "
+        "Attempting to download recipe from "
+        f"{HUGGINGFACE_CO_URL_HOME}"
+    )
+    try:
+        _LOGGER.info(
+            f"Found recipe: {recipe_name} for model id: "
+            f"{model_path}. Downloading..."
+        )
+        recipe = hf_hub_download(repo_id=model_path, filename=recipe_name)
+    except Exception as e:
+        _LOGGER.info(
+            f"Unable to to find recipe {recipe_name} "
+            f"for model id: {model_path}: {e}. "
+            "Skipping recipe resolution."
+        )
+        recipe = None
+    return recipe, True
+
+
+def recipe_from_sparsezoo_stub(
+    stub: str, recipe_file_name: Optional[str] = None
+) -> Optional[str]:
+    """
+    Attempts to find the recipe for the sparsezoo stub.
+
+    :param stub: The sparsezoo stub to find the recipe for
+    :param recipe_file_name: The name of the recipe file to find.
+        If None, the default recipe will be returned. Default: None
+    :return: The path to the recipe file if found, None otherwise
+    """
+    if recipe_file_name is None:
+        recipe = Model(stub).recipes.default.path
+        _LOGGER.info(f"Found recipe: {recipe}")
+        return recipe
+    else:
+        for recipe in Model(stub).recipes.recipes:
+            if recipe.name == recipe_file_name:
+                recipe - recipe.path
```

Comment:
```suggestion
                recipe - recipe.path
```
What does this do? 👀

Reply:
should be = hehe